src/rebar3_hank_prv.erl

```diff
@@ -120,6 +127,28 @@ format_result(#{file := File,
                 text := Msg}) ->
     hank_utils:format_text("~ts:~tp: ~ts", [File, Line, Msg]).
 
+-spec maybe_write_data_to_json_file([hank_rule:result()], rebar_state:t()) -> ok.
+maybe_write_data_to_json_file(Result, State) ->
+    {Args, _} = rebar_state:command_parsed_args(State),
+    ConvertedResult = convert_data_to_binary(Result),
+    EncodedResult = jsx:encode(ConvertedResult),
```

Comment:
It's not terribly wrong, but I would prefer to only do this _if we're going to use it_ (i.e. First try to come up with the name of the output file, if it's not empty,... Only then convert, encode, and emit the output).

Reply:
And please document the rebar.config option in the README.md